### ISSUE TITLE
`TextBox` visual state is not updated when `IsEnabled` changes

### ISSUE THREAD

**Issue by jeromelaban** [unoplatform/uno#7791]
Repro: [7787-Uno.Client.zip](https://github.com/unoplatform/uno/files/7846850/7787-Uno.Client.zip)

### Discussed in https://github.com/unoplatform/uno/discussions/7787

<div type='discussions-op-text'>

<sup>Originally posted by **AnuragBhatt** January 10, 2022</sup>
I am trying to modify the isEnabled property of the textbox dynamically as shown in the code below: 
<TextBox Name="SearchTxt" KeyDown="SearchText_Keydown"
       IsEnabled="{x:Bind ViewModel.IsSearchEnabled, Mode=OneWay}"
Text="{x:Bind ViewModel.SearchText, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}">
                </TextBox>

However, it behaves quite strangely. 

ViewModel.IsSearchEnabled being a Boolean is initially false by default.
After a while in page load, I am setting it to true in View Model.
Now it is shows up in the UI as if it is still disabled (looking at background color).
 (Picture 1)
But if I try to hover and click on it, it then changes the color to default white (and actually I am able to click upon it, so that means IsEnabled has worked).
 
So this behaviour is only observed when I try to bind the “IsEnabled” attribute to some Viewmodel property.
If I try to hardcode IsEnabled as true or false, everything works just fine.

Style triggers won’t also work in UWP/WinUI world, but they used to work in WPF.
Not sure about VisualStateManager triggers.
Could somebody please point me how to get past this issue?
This is causing UI to not behave properly in many of the places.

code is shared at this location:
https://github.com/AnuragBhatt/Uno.Client

Steps to run the code:

1. download the code from repo
2. Make Uno.Client.Wasm as startup project.
3. Run the project.
4. Initially you will see the textbox disabled with "Initial Text" as content (try clicking on it to ensure that it is disabled)
5. Click on the button below, that will change the text of the text box and will enable it.
6. Problem found: the background color style of the textbox still gives the visual effect of a disabled textbox whereas it should be enabled and shown in white.
7. if you hover upon the textbox, it magically changes the background as desired.
</div>

**Comment by AnuragBhatt** [comment]
Trying to attach zip file here but it is giving error message like below:
![image](https://user-images.githubusercontent.com/8007980/149037805-d417569b-1d40-4c29-ba6a-d7769de5edb0.png)
Is there any limit to the size of zip file?

**Comment by Youssef1313** [comment]
https://github.com/unoplatform/uno/blob/acfac1b01148869993e3fcc7cb3a6755b1ee9ac1/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs#L108-L113

I think there is a missing `UpdateVisualState()` call. It might not be Wasm only. This missing call should be in a platform-independent code path, ie, have the following code in `TextBox.cs`:

```csharp
 private protected override void OnIsEnabledChanged(IsEnabledChangedEventArgs e) 
 { 
 	base.OnIsEnabledChanged(e); 
 	UpdateVisualState();
 	OnIsEnabledChangedPartial();
 }

partial void OnIsEnabledChangedPartial(); // platform-specific implementations should implement this partial method.
```

**Comment by AnuragBhatt** [comment]
Thanks @Youssef1313 for the update! Is this a fix to be done at UNO end? If so, when is that bugfix getting released?

**Comment by Youssef1313** [comment]
@AnuragBhatt Yes it should be on Uno. There is no PR yet. You can open a PR yourself to fix it if you want to [contribute](https://platform.uno/docs/articles/uno-development/contributing-intro.html), or wait until someone else picks it up.

**Comment by Alex-DE-74** [comment]
Hello!
Is any fix as uno-update available?
Please check IsReadonly also. Seems to be buggy also. Maybe it affects all states?

**Comment by ajpinedam** [comment]
> Hello! Is any fix as uno-update available? Please check IsReadonly also. Seems to be buggy also. Maybe it affects all states?

Hi. There is a PR (https://github.com/unoplatform/uno/pull/7930) with the fix already. This PR will be merged soon. 

Re: `IsReadOnly`: What kind of issue are you seeing with this property? I can take a look at it.

**Comment by ajpinedam** [comment]
Tested the `IsReadonly` with the PR/Fix on #7930  applied, and it looks good to me.

**Comment by AnuragBhatt** [comment]
Thank you very much @jeromelaban and the UNO contributors, I have fetched the latest package and confirm this issue is no more there..! Appreciate your efforts on the same.
